### PR TITLE
Added cron for dumping JS Routing

### LINKED ISSUE #21200: CD: Dump JS routing
#### Is your feature request related to a problem?

Some actions are performed too late in the build process, resulting in delays. We can leverage CI to automate these actions nightly.

#### Describe the solution you'd like

Dump the JS routing to ensure it's always up to date, then [create an automated Pull Request](https://peterevans.dev/posts/github-actions-how-to-create-pull-requests-automatically/) using the @prestonBot user.

```
php bin/console fos:js-routing:dump --format=json --target=admin-dev/themes/new-theme/js/fos_js_routes.json
```

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Added cron for dumping JS Routing
| Type?             | improvement
| Category?         | PM
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #21200
| How to test?      | N/A

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22588)

### CODE REVIEW

**Review by PierreRambaud — DISMISSED** [review]

**Review by jolelievre — COMMENTED** [review]
I have a suggestion, instead of building this via a cron Why not manage it like other assets: remove it from the repository and build it via `make assets` This way we don't need to update it

**Comment by PierreRambaud** [comment]
> I have a suggestion, instead of building this via a cron Why not manage it like other assets: remove it from the repository and build it via `make assets` This way we don't need to update it

I tried, but we can't.
For building assets we need this file, otherwise the compilation is failing.
For running PrestaShop commands we need to have PrestaShop installed.
To install PrestaShop, we need assets.
Loop :trollface: 

We have a lot of work to make it possible :(

**Review by atomiix — DISMISSED** [review]

File: .github/workflows/cron.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v3
+        with:
+            token: ${{ secrets.GITHUB_TOKEN }}
+            commit-message: Updated JS Routing file
+            title: 'Updated JS Routing file'
+            body: >
+                    | Questions         | Answers
+                    | ----------------- | -------------------------------------------------------
```
Comment (matks) [inline .github/workflows/cron.yml]: Maybe it's not important to fill this template. The template aims to carry data inside the head of the author of the PR to maintainers, in order to help reviewing the PR. This one will be straightforward and "author" cannot answer if there are feedbacks.

Also it means everytime we update the PR template we have to update this GitHub Action

**Review by PierreRambaud — DISMISSED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thank you @Progi1984

**Pull request merged**